units/ftdc_rollups.go

```diff
@@ -55,19 +58,32 @@ func (j *ftdcRollupsJob) validate() error {
 		return errors.New("no artifact info given")
 	}
 
+	if len(j.RollupTypes) == 0 {
+		return errors.New("no rollup factories given")
+	}
+
 	return nil
 }
 
-func NewFTDCRollupsJob(perfId string, artifactInfo *model.ArtifactInfo) (amboy.Job, error) {
+func NewFTDCRollupsJob(perfId string, artifactInfo *model.ArtifactInfo, factories []perf.RollupFactory, user bool) (amboy.Job, error) {
 	j := makeFTDCRollupsJob()
 	j.PerfID = perfId
 	j.ArtifactInfo = artifactInfo
+	j.UserSubmitted = user
+
+	j.RollupTypes = []string{}
+	for _, factory := range factories {
+		// TODO: Should we add an assertion here, making sure that the
+		// rollup types are in the "registry"? Can also do this in
+		// validate.
```

Comment:
probably!

Reply:
So in the Run command I continue on error if I cannot resolve a specific factory. Do you think it is a better idea to fail early and not run anything at all?

Reply:
at runtime is probably better, I'd just delete this comment.